draft-ietf-httpbis-semantics-latest.xml

```diff
@@ -6896,72 +6896,88 @@ Accept-Language: da, en-gb;q=0.8, en;q=0.7
    <x:anchor-alias value="Vary"/>
 <t>
    The "Vary" header field in a response describes what parts of a request
-   message, aside from the method and target URI,
-   might influence the origin server's process for selecting and representing
-   this response.
+   message, aside from the method and target URI, might have influenced the
+   origin server's process for selecting the content of this response.
 </t>
 <sourcecode type="abnf7230"><iref primary="true" item="Grammar" subitem="Vary"/>
   <x:ref>Vary</x:ref> = #( "*" / <x:ref>field-name</x:ref> )
 </sourcecode>
 <t>
-   A Vary field value is a list of request field names, known as the selecting header fields,
-   that might have a role in selecting the representation for this response. Potential selecting
-   header fields are not limited to those defined by this specification.
+   A Vary field value is either the wildcard member "*" or a list of
+   request field names, known as the selecting header fields, that might
+   have had a role in selecting the representation for this response.
+   Potential selecting header fields are not limited to fields defined by
+   this specification.
 </t>
 <t>
-   If the list contains "*", it signals that other aspects of the request might
-   play a role in selecting the response representation, possibly including
-   elements outside the message syntax (e.g., the client's network address).
+   A list containing the member "*" signals that other aspects of the
```

Comment:
...but this admits the possibility of a list containing `*` (and other members).

Reply:
Yes, that's because the ABNF now allows a list containing more than one *, or a mix of names and *. The reason for that is it allows Vary field lines to be appended multiple times. Regardless, the impact is the same for a recipient if they parse according to the ABNF and process as specified in "containing a member".

The way this is phrased is deliberate, since for any given Vary field line there is no reason to send both a * and field names, so we do want to discourage that even if it is allowed by the grammar for other reasons.

Reply:
Thank you for confirming.